sdk/identity/identity/src/credentials/authFileCredential.ts

```
@@ -0,0 +1,91 @@
+   */
+  constructor(filePath: string) {
+    let authData = JSON.parse(fs.readFileSync(filePath).toString());
+    if (this.credential == null && typeof authData == "object") {
+      let clientId = authData["clientId"];
+      let certificatePath = authData["certificatePath"];
+      let tenantId = authData["tenantId"];
+      let activeDirectoryEndpointUrl = authData["activeDirectoryEndpointUrl"];
+
+      if (
+        clientId == undefined ||
+        certificatePath == undefined ||
+        tenantId == undefined ||
+        activeDirectoryEndpointUrl == undefined
+      ) {
+        throw new Error("there was a problem building the credential.");
+      }
+
+      this.credential = new ClientCertificateCredential(tenantId, clientId, certificatePath, {
+        authorityHost: activeDirectoryEndpointUrl
+      });
+    }
+  }
+
+  /**
+   * Authenticates with Azure Active Directory and returns an access token if
+   * successful.  If authentication cannot be performed at this time, this method may
+   * return null.
+   *
+   * @param scopes The list of scopes for which the token will have access.
+   * @param options The options used to configure any requests this
+   *                TokenCredential implementation might make.
+   */
+  async getToken(
+    scopes: string | string[],
+    options?: GetTokenOptions
+  ): Promise<AccessToken | null> {
+    const { span, options: newOptions } = createSpan("authFileCredential-getToken", options);
+    if (this.credential) {
+      try {
+        return await this.credential.getToken(scopes, newOptions);
+      } catch (err) {
+        const code =
+          err.name === AuthenticationErrorName
+            ? CanonicalCode.UNAUTHENTICATED
+            : CanonicalCode.UNKNOWN;
+        span.setStatus({
+          code,
+          message: err.message
+        });
+        throw err;
+      } finally {
+        span.end();
+      }
+    }
```

Should this throw an error rather than returning null?